src/Microsoft.AspNetCore.Server.IISIntegration/WebHostBuilderIISExtensions.cs

```diff
@@ -39,6 +39,22 @@ public static IWebHostBuilder UseIISIntegration(this IWebHostBuilder hostBuilder
                 return hostBuilder;
             }
 
+            // Check if in process
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows) && NativeMethods.is_ancm_loaded())
+            {
+                hostBuilder.UseSetting(nameof(UseIISIntegration), true.ToString());
+                hostBuilder.CaptureStartupErrors(true);
+
+                // TODO put this in options and use path.
+                var applicationPath = NativeMethods.http_get_application_full_path();
+                hostBuilder.UseContentRoot(applicationPath);
+                return hostBuilder.ConfigureServices(services =>
+                {
+                    services.AddSingleton<IServer, IISHttpServer>();
+                    services.AddAuthenticationCore();
```

Comment:
Why are we adding authentication?

Reply:
It's something added by the other IISIntegration code path and we'll need it for Windows Auth anyways.

Reply:
Will AddAuthenticationCore noop if we don't have authentication?

Reply:
It's only registering passive services, it's fine.